File: 0226-invert-binary-tree/0226-invert-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
*/

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        TreeNode* answer = root;
        preOrder(root);
        return answer;
    }

    void preOrder(TreeNode* root){
        if (root==nullptr) return;
        TreeNode* tempRoot = root->left;
        root->left = root->right;
        root->right = tempRoot;
        preOrder(root->left);
        preOrder(root->right);
    }
};

File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.invert(root)
        return root
        
    def invert(self,root):
        
        if root == None:
            return

        self.invert(root.left)
        self.invert(root.right)
        root.left, root.right = root.right, root.left